atc/api/accessor/accessor_factory.go

```diff
@@ -52,3 +72,51 @@ func (a *accessFactory) validate(token *jwt.Token) (interface{}, error) {
 
 	return a.publicKey, nil
 }
+
+type customRolesMapping map[string][]string
+
+func (a *accessFactory) CustomizeActionRoleMap(logger lager.Logger, configFilePath string) error {
+	if configFilePath == "" {
+		return nil
+	}
+
+	content, err := ioutil.ReadFile(configFilePath)
```

Comment:
One last thing :), can we do all our file reading at the atccmd level instead? That way the accessorFactory can simply take a map of the custom roles, and it makes the tests a lot simpler.

Reply:
OK, I added a new method `accessor.ParseCustomActionRoleMap()` and called it from `atc/atccmd/command.go`.